lib/sparrow/pools_warden.ex

```diff
@@ -94,33 +94,43 @@ defmodule Sparrow.PoolsWarden do
       end)
   end
 
-  @spec handle_info(any, :ok) :: {:noreply, :ok}
-  def handle_info(unknown, _state) do
+  @impl GenServer
+  def handle_info({:DOWN, _ref, :process, pid, reason}, state) do
+    {pid, pool_type, pool_name, tags} = List.keyfind(state, pid, 0)
+    :ets.delete_object(@tab_name, {pool_type, {pool_name, tags}})
+
+    _ =
+      Logger.info(fn ->
+        "worker=pools_warden, action=unregistering pool, pool_name=#{inspect(pool_name)}, pid=#{inspect(pid)},
+        reason=#{inspect(reason)}"
+      end)
+
+
+    {:noreply, List.delete(state, {pid, pool_type, pool_name, tags})}
+  end
+
+
+  def handle_info(unknown, state) do
```

Comment:
Do you want to crush on unknown messages or log them and curry on?
Both ideas are correct ;)

Reply:
Log them and carry on - that's how it works now and how it worked before - there are no changes here